diff --git a/drivers/w1/w1_zephyr_gpio.c b/drivers/w1/w1_zephyr_gpio.c
index 33d6d766a46..79116749635 100644
--- a/drivers/w1/w1_zephyr_gpio.c
+++ b/drivers/w1/w1_zephyr_gpio.c
@@ -300,6 +300,9 @@ static int w1_gpio_init(const struct device *dev)
 		return -ENODEV;
 	}
 
+	/* Zero-initialization does not initialize the mutex's wait queue. */
+	k_mutex_init(&data->master_data.bus_lock);
+
 	data->timing = &std;
 	data->overdrive_active = false;
 
